athenet/sparsifying/utils/interval.py

```diff
@@ -0,0 +1,284 @@
+"""Intervals in Theano including special functions for sparsifying."""
+
+from athenet.sparsifying.utils.numlike import Numlike
+from theano import function
+from theano import tensor as T
+from theano.ifelse import ifelse
+
+
+class Interval(Numlike):
+    """Theano interval class
+
+    Note: Should be treated as interval type with bounds as Theano nodes.
+    Opetations on Interval create nodes in Theano graph."""
+
+    def __init__(self, lower, upper):
+        """Creates interval.
+
+        :lower: lower bound of Interval to be set
+        :upper: upper bound if Interval to be set
+
+        Note: lower must be lower than upper. It is not being checked."""
```

Comment:
I think it should not. Note that such check must be included in Theano graph. Then, as __init__ is basic interval matrix operation, check would be executed after every creation of new node in Theano graph. Therefore we could almost double size of the graph. What is worse we will probably prevent Theano from making optimizations of graph because every node would be significant and checked. How about this?